Coupled-Cluster/spin_free_CC/helper_cc.py

```diff
@@ -1579,6 +1580,192 @@ def linresp(self):
+        self.t2 = ccsd.t2
+
+        # Steal "ova" translation
+        self.slice_nfzc = cchbar.slice_nfzc
+        self.slice_o = cchbar.slice_o
+        self.slice_v = cchbar.slice_v
+        self.slice_a = cchbar.slice_a
+        self.slice_dict = cchbar.slice_dict
+
+        # Steal Hbar blocks
+        self.Hov = cchbar.Hov
+        self.Hoo = cchbar.Hoo
+        self.Hvv = cchbar.Hvv
+        self.Hoooo = cchbar.Hoooo
+        self.Hvvvv = cchbar.Hvvvv
+        self.Hvovv = cchbar.Hvovv
+        self.Hooov = cchbar.Hooov
+        self.Hovvo = cchbar.Hovvo
+        self.Hovov = cchbar.Hovov
+        self.Hvvvo = cchbar.Hvvvo
+        self.Hovoo = cchbar.Hovoo
+
+        # Steal L integrals (L[pqrs] = 2*MO[pqrs] - MO[pqsr])
+        self.L = cchbar.L
+
+        # Build Approximate Diagonal of Hbar
+        self.Dia = self.Hoo.diagonal().reshape(-1, 1) - self.Hvv.diagonal()
+        self.Dijab = self.Hoo.diagonal().reshape(
+            -1, 1, 1, 1) + self.Hoo.diagonal().reshape(
+                -1, 1, 1) - self.Hvv.diagonal().reshape(
+                    -1, 1) - self.Hvv.diagonal()
+
+    def get_MO(self, string):
+        if len(string) != 4:
+            psi4.core.clean()
+            raise Exception('get_MO: string %s must have 4 elements.' % string)
+        return self.MO[self.slice_dict[string[0]], self.slice_dict[string[1]],
+                       self.slice_dict[string[2]], self.slice_dict[string[3]]]
+
+    def get_F(self, string):
+        if len(string) != 2:
+            psi4.core.clean()
+            raise Exception('get_F: string %s must have 4 elements.' % string)
+        return self.F[self.slice_dict[string[0]], self.slice_dict[string[1]]]
+
+    def get_L(self, string):
+        if len(string) != 4:
+            psi4.core.clean()
+            raise Exception('get_L: string %s must have 4 elements.' % string)
+        return self.L[self.slice_dict[string[0]], self.slice_dict[string[1]],
+                      self.slice_dict[string[2]], self.slice_dict[string[3]]]
+
+    def build_sigma1(self, C1, C2):
+        """
+        Compute the contributions to <ia|Hbar*C|0>
+
+        Parameters
+        ----------
+        C1: array like, shape(ndocc, nvir)
+          The first nsingles elements of a guess vector reshaped to (o,v)
+
+        C2: array like, shape(ndocc,ndocc,nvir,nvir)
+          The last ndoubles elements of a guess vector reshaped to (o,o,v,v)
+
+        Returns
+        -------
+        S1: ndarray shape(ndocc, nvir)
+
+        Examples
+        --------
+
+        >>> # Get some vectors as cols of a 2D numpy array and orthogonalize them
+        >>> c  = np.random.rand(eom.nsingles + eom.ndoubles, 2)
+        >>> c,  = np.linalg.qr(c)
+
+        >>> # Slice out the singles, doubles blocks of the first vector and reshape
+        >>> C1 = c[:,:nsingles].reshape(eom.ndocc, eom.nvir)
+        >>> C2 = c[:,nsingles:].reshape(eom.ndocc, eom.ndocc, eom.nvir, eom.nvir)
+        >>> S1 = eom.build_sigma1(C1, C2)
```

Comment:
This class looks awesome, with docs! Can you add the original papers that you pull from in the `__init__` as well?

Reply:
Sure, you mean the docstring for `__init__` right?

Reply:
Should that be a "References" section, "See Also" section, notes section, or something else?

Reply:
Yea, the `__init__` docstring. Please use the canonical `Notes` section for this docstring.

Reply:
Do you want me to follow [this](https://github.com/numpy/numpy/blob/master/doc/example.py#L98-L108) style for the references (from the numpy docs examples) just put them under a `Notes` heading?

Reply:
Just put them under notes. I think picking up references like that requires some special tech.